lib/offline.js

```diff
@@ -110,6 +111,15 @@ module.exports = function(config) {
       }, {}));
     }
 
+    function getPathName(filepath, rootDir, directoryIndexes) {
+       filepath = filepath.replace(new RegExp('/(' + directoryIndexes.join('|') + ')$'), '/');
```

Comment:
This regex constructor call should escape the values of *directoryIndexes*, so the regex only matches their literal values. It may not matter for the common case, because *index.html* isn't likely to match anything else (although the period is a special character, so it *could* do so); but it seems like a footgun, likely to fail in unexpected ways that are challenging to diagnose.

The https://www.npmjs.com/package/escape-string-regexp module seems like it'll do the job, and its interface is compatible with *Array.map*, so this should be as simple as `directoryIndexes.map(escapeStringRegexp).join('|')` (although I haven't tested that yet!).

Also, since the regex won't change over the lifetime of the call to the *module.exports* outer function, you can theoretically optimize by creating the regex once and then reusing it for each call to *getPathName*. Although I'm not sure how much difference that makes in practice, since I think JS engines know how to optimize it away these days.

Either way, you don't need to pass *directoryIndexes* into *getPathName*, since it has access to that variable via the outer function, whose scope it closes over.